### PR TITLE
Adds expected execution time for delayed jobs & new promote job functionality

### PR DESCRIPTION
**Pull request by benbenedek**
image

### CODE REVIEW

**Comment by benbenedek** [comment]
@skeggse Not sure who should I tag to review this PR, WDYT? 

For our use case of arena / Bull, we find this functionality crucial.

**Comment by sibelius** [comment]
related to this one https://github.com/bee-queue/arena/pull/166?

**Comment by benbenedek** [comment]
@sibelius looks very similar, only missing the "Promote Job" functionality.

**Comment by matthiasfeist** [comment]
Great PR! This would also be a function that I really need!

**Comment by benbenedek** [comment]
Are we planning to merge this & publish version?

My workaround is to consume from my own repo, but I believe this is required functionality for us to be using your awesome tool.

Thanks for your hard work at this project.

**Comment by bvallelunga** [comment]
@ttacon any updates on merging this PR?

File: src/server/views/helpers/handlebars.js [hunk 1]
```diff
@@ -42,6 +42,10 @@ const helpers = {
 
   hashIdAttr(id) {
     return crypto.createHash('sha256').update(id).digest('hex');
+  },
+
+  getDelayDate(opts) {
+    return opts.timestamp + opts.delay;
```
Comment (skeggse) [inline src/server/views/helpers/handlebars.js:48]:
```suggestion
  getDelayDate(job) {
    return job.timestamp + job.delay;
```

Comment (skeggse) [inline src/server/views/helpers/handlebars.js:48]: In bee-queue, these fields don't exist. Instead, `job.options.delay` is the timestamp at which the job will resume.

File: public/dashboard.js [hunk 2]
```diff
@@ -30,6 +30,29 @@ $(document).ready(() => {
...
+    const queueHost = $(this).data('queue-host');
+
+    const r = window.confirm(`Promote #${jobId} in queue "${queueHost}/${queueName}"?`);
+    if (r) {
+      $.ajax({
+        method: 'PUT',
+        url: `${basePath}/api/queue/${encodeURIComponent(queueHost)}/${encodeURIComponent(queueName)}/job/${encodeURIComponent(jobId)}`
+      }).done(() => {
+        window.location.reload();
+      }).fail((jqXHR) => {
```
Comment (skeggse) [inline public/dashboard.js:47]: What do you think about doing `$(this).prop('disabled', false);` in the failure case here, too?

**Comment by tevaum** [comment]
@benbenedek, are you looking at the change requests?

**Comment by skeggse** [comment]
Closing due to inactivity. Other contributors are likely welcome to pick this up and address the compatibility concerns.

**Pull request closed without merging**